lib/rjgit_adapter/git_layer_rjgit.rb

```diff
@@ -160,11 +160,14 @@ def grep(query, options={})
           blobs << RJGit::Blob.new(@git.jrepo, item[:mode], item[:path], walk.lookup_blob(ObjectId.from_string(item[:id]))) if item[:type] == 'blob'
         end
         result = []
+        regex_query = query.scan(/"([^"]+)"|(\S+)/).flatten.compact.map {|word| Regexp.escape(word)}
+        regex_query = "(?:#{regex_query.join('|')})"
```

Comment:
It would be cleaner I think if this `regex_query` variable would already include the bits that you later add in `blob.data.scan` (i.e., `^(.*` etc). That cleans up the `scan` call.

Reply:
👍

I wonder now, though, whether it would no be better to keep the `grep` method in the adapter the same, and do all this constructing of search terms/regex in `gollum-lib`. That would prevent code duplication in the rugged/rjgit adapters.

Reply:
> do all this constructing of search terms/regex in gollum-lib.

Even better. 👍